vissl/data/ssl_dataset.py

```diff
@@ -221,7 +234,53 @@ def _load_labels(self):
                 raise ValueError(f"unknown label source: {label_source}")
             self.label_objs.append(labels)
 
-    def __getitem__(self, idx):
+    def _init_image_and_label_subset(self):
+        """
+        If DATA_LIMIT = K >= 0, we reduce the size of the dataset from N to K.
+
+        This function will create a mapping from [0, K) to [0, N), using the
+        parameters specified in the DATA_LIMIT_SAMPLING configuration. This
+        mapping is then cached and used for all __getitem__ calls to map
+        the external indices from [0, K) to the internal [0, N) indices.
+
+        This function makes the assumption that there is one data source only
+        or that all data sources have the same length (same as __getitem__).
+        """
+
+        valid_datasets = {
```

Comment:
I think we should move this check down to line 305 . We should just not call the init_subset function if the conditions are not met.

Reply:
It should be fixed at 5586fac !